scheduler/src/cook/scheduler/constraints.clj

```diff
@@ -108,14 +108,16 @@
   (job-constraint-evaluate
     [{:keys [job]} _ vm-attributes _]
     (let [k8s-vm? (= (get vm-attributes "compute-cluster-type") "kubernetes")
-          job-gpu-count-requested (-> job util/job-ent->resources :gpus (or 0))]
+          ; cast to int because the API allows both float and int gpu values
+          job-gpu-count-requested (-> job util/job-ent->resources :gpus (or 0) int)]
           (if k8s-vm?
             (let [job-gpu-model-requested (job->gpu-model-requested
                                             job-gpu-count-requested job (util/job->pool-name job))
                   vm-gpu-model->count-available (get vm-attributes "gpus")
                   vm-satisfies-constraint? (if (pos? job-gpu-count-requested)
                                              ; If job requests GPUs, require that the VM has enough gpus available in the same model as the job requested.
-                                             (>= (get vm-gpu-model->count-available job-gpu-model-requested 0) job-gpu-count-requested)
+                                             ; cast to int because the API allows both float and int gpu values
+                                             (= (int (get vm-gpu-model->count-available job-gpu-model-requested 0)) job-gpu-count-requested)
```

Comment:
is it better style to use (== ...) ?

Reply:
yes thank you! didn't know that existed but that is definitely cleaner and I modified this